modules/config/schedule_rotations.tf

```diff
@@ -0,0 +1,54 @@
+resource "opsgenie_schedule_rotation" "this" {
+  for_each = module.this.enabled ? { for schedule_rotation in local.schedule_rotations : schedule_rotation.name => schedule_rotation } : tomap()
+
+  name = each.value.name
+  type = each.value.type
+
+  start_date = each.value.start_date
+  end_date   = try(each.value.end_date, null)
+  length     = try(each.value.length, null)
+
+  # Look up our schedule id by name
+  schedule_id = try(opsgenie_schedule.this[each.value.schedule_name].id, null)
+
+  dynamic "participant" {
+    for_each = try(each.value.participants, [])
+
+    content {
+      type = participant.value.type
+      id   = data.opsgenie_user.this[participant.value.username].id
+    }
+  }
+
+  dynamic "time_restriction" {
+    for_each = try(each.value.time_restriction, null) != null ? ["true"] : []
+
+    content {
+      type = each.value.time_restriction.type
```

Comment:
```suggestion
      type = time_restriction.type
```

Reply:
@aknysh  `type` is the only "global" value under `time_restriction` and is only specified a single time.. there are some settings below the time_restriction that may repeat (set values), for `restriction` or `restrictions` (depending on the type passed).